src/test/java/e2e/pages/BasePage.java

```diff
@@ -0,0 +1,46 @@
+package e2e.pages;
+
+import e2e.WebDriverProvider;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+
+public abstract class BasePage {
+
+    protected final String BASE_PATH = "http://localhost:9000/#/";
+    protected String path;
+    protected WebDriver driver;
+    protected String uniqueId;
```

Comment:
This way of implementing the BasePage doesn't force a developer who will extend this to fill this field out. In my opinion, we should make it more apparent - maybe instead of protected fields we can add abstract getters that will need to be implemented by extending classes? This will force everybody to properly handle this mechanism.